client/src/ducks/authentication/authentication.js

```diff
@@ -0,0 +1,140 @@
+import jwtDecode from 'jwt-decode';
+
+import auth from '../../singletons/authentication';
+import history from '../../singletons/history';
+import config from '../../config';
+import { willExpireAt } from '../../utils/utils';
+import { actions as userActions } from '../user/user';
+
+export const types = {
+  LOGIN_REQUEST: 'AUTHENTICATION/LOGIN_REQUEST',
+  LOGIN_SUCCESS: 'AUTHENTICATION/LOGIN_SUCCESS',
+  LOGIN_FAILURE: 'AUTHENTICATION/LOGIN_FAILURE',
+  HANDLE_AUTHENTICATION: 'AUTHENTICATION/HANDLE_AUTHENTICATION',
+  LOGOUT: 'AUTHENTICATION/LOGOUT'
+};
+
+export const actions = {
+  loginRequest: () => {
+    auth.authorize();
+    return {
+      type: types.LOGIN_REQUEST
+    };
+  },
+  loginSuccess: (accessToken, expiresIn, userInfo) => ({
+    type: types.LOGIN_SUCCESS,
+    accessToken,
+    expiresIn,
+    userInfo
+  }),
+  loginFailure: error => ({
+    type: types.LOGIN_FAILURE,
+    error
+  }),
+  handleAuthentication: () =>
+    // redux-thunk knows to handle action functions instead of normal action objects
+    // this allows us to dispatch actions within actions, as well as access state
+    (dispatch, getState) => {
+      dispatch({
+        type: types.HANDLE_AUTHENTICATION
+      });
+
+      // shorthand for when we encounter an error during this authentication process
+      const errorOut = err => {
+        console.error(err);
+        dispatch(actions.loginFailure(err));
+        // TODO - go somewhere for failure or unauthenticated jawns
+        history.replace('/');
+      };
+
+      auth.parseHash((err, authResult) => {
+        if (authResult && authResult.accessToken && authResult.idToken) {
+          try {
+            const decodedIdToken = jwtDecode(authResult.idToken);
+            console.log(decodedIdToken);
```

Comment:
Oops, gonna take these `logs` out. Also need to submit an issue about adding a robust client-side logger